core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1368,6 +1368,14 @@ package object config {
         s"The buffer size must be greater than 0 and less than or equal to ${Int.MaxValue}.")
       .createWithDefault(4096)
 
+  private[spark] val SHUFFLE_CHECKSUM =
+    ConfigBuilder("spark.shuffle.checksum")
```

Comment:
This actually follows the naming of `spark.shuffle.compress`.

(At that time, I don't have any other configs. But now, I'm thinking of making the diagnosis optional with a conf. So, it might be reasonable to append ".enabled".)

Do you have any other configs in mind?

Reply:
`algorithm`, for now, could be another in the namespace.

Reply:
Appened `.enabled` and added `spark.shuffle.checksum.algorithm` as well.